src/test/fixtures/mocks/request-animation-frame.js

```diff
@@ -21,15 +21,35 @@
  *
  */
 export default function mockRaf() {
-  let fns = [];
+  const fns = [];
   jest.spyOn(window, 'requestAnimationFrame').mockImplementation(fn => {
     fns.push(fn);
   });
 
-  return () => {
-    for (const fn of fns) {
-      fn();
+  /**
+   * Execute the queued functions, one at a time.
+   *
+   * If no argument is passed, flush and execute all functions in the
+   * queue.
+   *
+   * If an array of timestamps is passed, those timestamps will be
+   * passed in sequence to each queued function. When the array runs
+   * out, the execution loop will stop, even if there are remaining
+   * functions to be called.
+   */
+  return (timestamps: ?(number[])) => {
+    while (fns.length > 0) {
+      let arg = undefined;
+      if (timestamps) {
+        arg = timestamps.shift();
+        if (arg === undefined) {
+          // We've run into the end of the passed array of
+          // timestamps. End the loop.
+          return;
```

Comment:
The reason for adding the ability to break out of the loop was that it would otherwise never exit when used with the `_keyboardNavigation` callback. That callback continues to request frames until the backing `_keysDown` set is empty. But that set is cleared with keyup events, and if the test is busy processing this loop the simulated keyup-event is never handled.

Reply:
I think this original function is technically incorrect. I believe it should create a copy of the original first:

```
export default function mockRaf() {
  let fns = [];
  jest.spyOn(window, 'requestAnimationFrame').mockImplementation(fn => {
    fns.push(fn);
  });

  return () => {
    const oldFns = fns;
    fns = [];
    for (const fn of oldFns) {
      fn();
    }
  };
}
```

Unfortunately I just ran it and it caused a few snapshot failures with existing tests. I'm a bit torn because this is a bit outside of the scope of the current work, but I think it would be better to have the simpler test fixture.

Reply:
I see. I thought the current behavior was intentional. I'll change it to the version you provide here. There are only two tests failing and they both can be fixed with an extra flush call.